src/libraries/System.Memory/tests/Span/EnumerateRunePositions.cs

```diff
@@ -0,0 +1,248 @@
+            {
+                Assert.Equal(expected, runePosition);
+                Assert.Equal(runePosition, expected);
+
+                Assert.True(expected.Equals(runePosition));
+                Assert.True(runePosition.Equals(expected));
+
+                Assert.True(((object)expected).Equals(runePosition));
+                Assert.True(((object)runePosition).Equals(expected));
+
+                Assert.True(expected == runePosition);
+                Assert.True(runePosition == expected);
+
+                Assert.False(expected != runePosition);
+                Assert.False(runePosition != expected);
+
+                Assert.Equal(expected.GetHashCode(), runePosition.GetHashCode());
+            }
+            else
+            {
+                Assert.NotEqual(expected, runePosition);
+                Assert.NotEqual(runePosition, expected);
+
+                Assert.False(expected.Equals(runePosition));
+                Assert.False(runePosition.Equals(expected));
+
+                Assert.False(((object)expected).Equals(runePosition));
+                Assert.False(((object)runePosition).Equals(expected));
+
+                Assert.False(expected == runePosition);
+                Assert.False(runePosition == expected);
+
+                Assert.True(expected != runePosition);
+                Assert.True(runePosition != expected);
+
+                Assert.NotEqual(expected.GetHashCode(), runePosition.GetHashCode());
+            }
+        }
+
+        private static void RunePosition_TestDeconstruct(RunePosition runePosition)
+        {
+            {
+                (Rune rune, int startIndex) = runePosition;
+                Assert.Equal(runePosition.Rune, rune);
+                Assert.Equal(runePosition.StartIndex, startIndex);
+            }
+            {
+                (Rune rune, int startIndex, int length) = runePosition;
+                Assert.Equal(runePosition.Rune, rune);
+                Assert.Equal(runePosition.StartIndex, startIndex);
+                Assert.Equal(runePosition.Length, length);
+            }
+        }
+
+        [Fact]
+        public static void RunePosition_DefaultTest()
+        {
+            RunePosition runePosition = default;
+            RunePosition_TestProps(default, 0, 0, false, runePosition);
+            RunePosition_TestEquals(default, runePosition);
+            RunePosition_TestDeconstruct(runePosition);
+
+            runePosition = new RunePosition();
+            RunePosition_TestProps(default, 0, 0, false, runePosition);
+            RunePosition_TestEquals(default, runePosition);
+            RunePosition_TestDeconstruct(runePosition);
+        }
+
+        [Theory]
+        [InlineData("x")]
+        [InlineData("\uD86D\uDF54")] // valid surrogate pair
+        [InlineData("\uFFFD")] // literal U+FFFD
+        [InlineData("\0")]
+        public static void RunePosition_UnitTest(string runeStr)
```

Comment:
This test only exercises a ctor which is supposed to be internal. I think it should be deleted.